### PR TITLE
Handle index overflow

### PR DESCRIPTION
**Pull request by mohammedabdon**
This change handles index overflow by comparing index against MaxUint64.
Also add a test for the overflow case.

### CODE REVIEW

File: api/layout/paths.go [hunk 1]
```diff
@@ -132,6 +133,9 @@ func ParseTileIndexWidth(index string) (uint64, uint64, error) {
 		if err != nil || n >= 1000 || len(indexPath) != 3 {
 			return 0, 0, fmt.Errorf("failed to parse tile index")
 		}
+		if n > (math.MaxUint64-n)/1000 {
```
[the remark this answers is not in the document]
Comment (mohammedabdon) [reply api/layout/paths.go]: oops, fixed now

**Review by phbnf — APPROVED** [review]
